db/columns/operations/select.py

```diff
@@ -151,6 +151,17 @@ def get_column_default_dict(table_oid, attnum, engine, metadata, connection_to_u
     return {"value": default_value, "is_dynamic": is_dynamic}
 
 
+def determine_whether_column_contains_data(
+        table_oid, column_name, engine, metadata, connection_to_use=None
+):
```

Comment:
Unclear if this is a predicate or not. I can see that the returned variable `contains_data` is a `.scalar()`, but I'd still struggle to parse whether the return is always meant to be a boolean. Maybe rename into `does_column_contain_data` or add a docstring describing return intent?

Reply:
I'll do the docstring. I prefer function names that make it quite obvious that they're a function in `__dict__` and so on.